### DISCUSSION TITLE
Switch Expression Block

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#7781] [Q&A] [answered]
Can you use blocks in switch expressions?
ie:
```csharp
var ret = input switch {
  _ => {
    return default;
  },
};
```

I feel like this should work but it doesnt seem to.

**Comment by iam3yal** [comment, marked as answer]
@TonyValenti It was suggested before, in fact I suggested it as well back then when they designed it and the response by Neal Gafter was to use local functions.

**Reply by HaloFour** [reply]
From what I recall the interest was in exploring sequence expressions in general since they would solve this case and others, but if that proved too problematic to explore as a specific syntax for `switch` expressions.  IMO, given that the most likely use case for it would be with `switch` I'd be in favor of ditching the larger sequence expression exploration.  I like the way the expression blocks with in Java, they use the `yield` keyword.

**Comment by 333fred** [comment]
Duplicate of https://github.com/dotnet/csharplang/issues/3037.

**Reply by TonyValenti** [reply]
I think I need to clarify something here.

In:
```csharp
var ret = input switch {
  _ => {
    return default;
  },
};
```

I intend the ```return``` statement to mean "set the result for this branch of the switch", not "return from the method".
It would have the same result as (although this might not be valid code):
```csharp
var ret = input switch {
  _ => () => {
    return default;
  }(),
};
```

**Reply by 333fred** [reply]
Yes, I understand that. That's why I linked the issue I did, as this is a duplicate (though I used `break` instead of `return` in my version).

**Comment by Unknown6656** [comment]
Yes please. Currently, I have to resort to solutions like:
```csharp
public static T Do<T>(Func<T> f) => f();

int x = ....;
int y = x switch {
    < 3 => 42,
    > 10 => 69,
    _ => Do<int>(() =>
    {
          // logic goes here
          return -1;
    }),
}
```
or even worse (readability-wise):
```csharp
int x = ....;
int y = x switch {
    < 3 => 42,
    > 10 => 69,
    _ => new Func<int>(() =>
    {
          // logic goes here
          return -1;
    })(),
}
```
